sdks/python/apache_beam/utils/counters.py

```diff
@@ -23,40 +23,69 @@
 For internal use only; no backwards-compatibility guarantees.
 """
 
+from collections import namedtuple
 import threading
+
 from apache_beam.transforms import cy_combiners
 
 
+"""Information identifying the IO being measured by a counter."""
+IOTargetName = namedtuple('IOTargetName', ['side_input_step_name',
+                                           'side_input_index',
+                                           'original_shuffle_step_name'])
+
+
+def side_input_id(step_name, input_index):
+  """Create an IOTargetName that identifies the reading of a side input."""
+  return IOTargetName(step_name, input_index, None)
+
+
+def shuffle_id(step_name):
+  """Create an IOTargetName that identifies a GBK step."""
+  return IOTargetName(None, None, step_name)
+
+
 class CounterName(object):
```

Comment:
YEs, it does. Updated to subclass namedtuple (to use default values and custom printing functions)